File: mobile-native-android/backend-api/src/main/java/com/edham/logistics/dto/UnifiedResponseDTO.java
package com.edham.logistics.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.Map;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UnifiedResponseDTO<T> {
    private Boolean success;
    private T data;
    private String message;
    private String error;
    private LocalDateTime timestamp;
    private Map<String, String> errors;
    private String requestId;
    private String version;
    private Map<String, Object> metadata;
}

File: mobile-native-android/app/src/main/java/com/edham/logistics/core/network/UnifiedResponseDto.kt
package com.edham.logistics.core.network

import com.google.gson.annotations.SerializedName

data class UnifiedResponseDto<T>(
    @SerializedName("success")
    val success: Boolean,
    
    @SerializedName("data")
    val data: T?,
    
    @SerializedName("message")
    val message: String?,
    
    @SerializedName("error")
    val error: String?,
    
    @SerializedName("timestamp")
    val timestamp: String?
)
